### ISSUE TITLE
Cannot find -l-lpthread

### ISSUE THREAD

**Issue by stangier** [ros-melodic-arch/ros-melodic-qt-gui-cpp#1]
When trying to install this packages during the build the following error occurs and the packages is not able to build successfully. 
```
[ 94%] Compiling generated code for qt_gui_cpp_sip Python bindings...
/usr/bin/ld: cannot find -l-lpthread
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:38: "/home/lorenz/dev/spare/qt_gui_core/qt_gui_cpp/build/devel/lib/python3.7/site-packages/qt_gui_cpp/libqt_gui_cpp_sip".so] Error 1
make[2]: *** [src/qt_gui_cpp_sip/CMakeFiles/libqt_gui_cpp_sip.dir/build.make:61: devel/lib/python3.7/site-packages/qt_gui_cpp/libqt_gui_cpp_sip.so] Error 2
make[1]: *** [CMakeFiles/Makefile2:380: src/qt_gui_cpp_sip/CMakeFiles/libqt_gui_cpp_sip.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

I've seen some people over at the [AUR](https://aur.archlinux.org/packages/ros-melodic-qt-gui-cpp/) having the same problem. It would be great if someone who has fixed it could reply.

**Comment by jwhendy** [comment]
This will be a meta comment about this project. Nothing wrong with your report, but the timing and goal of this project might help to explain where this falls in priorities.

- our goal is to get ROS building on arch
- the most [immediate goal](https://github.com/orgs/ros-melodic-arch/projects/1) is to get away from the horrible dependency on [ros-gbp](https://github.com/ros-gbp), where you only get releases/fixes when they decide to rebuild all of those sub-packages.
- in the meantime, [checksums will be broke to hell](https://github.com/ros-melodic-arch/ros-melodic-tf2-geometry-msgs/issues/1). Every time upstream rebuilds, we need to update the checksum. That's all it is. There are more than 350 packages currently being worked on super part time by two of us... that's a crapton of shasums to update.
- lastly, for sure there are *actual* issues beyond the mechanics of maintaining AUR packages for ROS that need to be addressed. For example the [signals issue]https://github.com/ros-melodic-arch/ros-melodic-tf2/pull/1/files). Note that many of these will be addressed above. For example, the proper way to do this is reporting the `signals` issue upstream. Note that it was [already done months ago](https://github.com/ros/geometry2/pull/354)... but ros-gbp lags, which is why we want to migrate all of these to ros on github.
- This is a by-product of that as well. Don't rely on AUR issues to understand these issues. Google until you find an issue from ROS alone. This `-lpthread` issue is related cmake and boost. Note that this [was reported](https://github.com/ros/ros-overlay/issues/711#issuecomment-455059118) (and I commented) a while ago too. Looks like it should have been fixed.

So... just wanted to put this into context. All the help we can get is appreciated, but I would wait on more comments until we get everything pointing upstream. A lot of this is just garbage from having no active maintainers for a really long time and pointing at a stale repo.

**Comment by jwhendy** [comment]
@bionade24 something like this would be good as a README for the repo. It's not a huge deal, but by the time we adjust the sources, we'll have many of these issues. Again, not a big deal to close, but also might be good to post the aim/steps/timeline for this project where others can see. Just an idea!

**Comment by bionade24** [comment]
The solution is pinned in the AUR. And I also wrote that I currently only support `ros-melodic-ros-base`, and that other packages won't work. For workarounds AUR comments are just fine. 
https://aur.archlinux.org/packages/ros-melodic-desktop-full/?O=10&PP=10

**Comment by stangier** [comment]
Okay, I did not know that only `ros-melodic-base` is supported by you. My goal is also to get ROS to build on Arch. Would it be possible to get some other form of communication going except through this issue? (IRC for example)

Thanks

**Comment by bionade24** [comment]
We used email the most time. But you can also contact me on freenode or keybase.
